From d77e5fc4bcdb7da748c9cca116a601ae4df60d21 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Tue, 31 May 2016 21:04:50 +0800
Subject: relaxng.c, xmlschemas.c: Fix build on pre-C99 compilers

Make sure that the variables are declared at the top of the block.

https://bugzilla.gnome.org/show_bug.cgi?id=767063
 ---
 relaxng.c    | 3 ++-
 xmlschemas.c | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/relaxng.c b/relaxng.c
index 56a3344..3d3e69c 100644
diff -ur misc/libxml2-2.9.4/relaxng.c misc/build/libxml2-2.9.4/relaxng.c
--- misc/libxml2-2.9.4/relaxng.c	2016-05-23 00:25:25.000000000 -0700
+++ misc/build/libxml2-2.9.4/relaxng.c	2016-08-28 00:37:26.017097000 -0700
@@ -2088,6 +2088,7 @@
                          const xmlChar * arg2)
 {
     char msg[1000];
+    xmlChar *result;
 
     if (arg1 == NULL)
         arg1 = BAD_CAST "";
@@ -2215,7 +2216,7 @@
         snprintf(msg, 1000, "Unknown error code %d\n", err);
     }
     msg[1000 - 1] = 0;
-    xmlChar *result = xmlCharStrdup(msg);
+    result = xmlCharStrdup(msg);
     return (xmlEscapeFormatString(&result));
 }
 
diff -ur misc/libxml2-2.9.4/xmlschemas.c misc/build/libxml2-2.9.4/xmlschemas.c
--- misc/libxml2-2.9.4/xmlschemas.c	2016-05-23 00:25:25.000000000 -0700
+++ misc/build/libxml2-2.9.4/xmlschemas.c	2016-08-28 00:37:26.025519000 -0700
@@ -3168,8 +3168,8 @@
 		"valid.");
 	}
 	if (expected) {
-	    msg = xmlStrcat(msg, BAD_CAST " Expected is '");
 	    xmlChar *expectedEscaped = xmlCharStrdup(expected);
+	    msg = xmlStrcat(msg, BAD_CAST " Expected is '");
 	    msg = xmlStrcat(msg, xmlEscapeFormatString(&expectedEscaped));
 	    FREE_AND_NULL(expectedEscaped);
 	    msg = xmlStrcat(msg, BAD_CAST "'.\n");
